src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs

```diff
@@ -125,10 +125,15 @@ protected override Size ArrangeOverride(Size finalSize)
 			}
 			else if (Popup.CustomLayouter == null)
 			{
+				if (Popup.PlacementTarget is not null)
+				{
+					return PlacementArrangeOverride(Popup, finalSize);
+				}
+				
 				// Gets the location of the popup (or its Anchor) in the VisualTree, so we will align Top/Left with it
 				// Note: we do not prevent overflow of the popup on any side as UWP does not!
 				//		 (And actually it also lets the view appear out of the window ...)
-				var anchor = Popup.Anchor ?? Popup;
+				var anchor = Popup.PlacementTarget ?? Popup;
```

Comment:
It looks like `Popup.PlacementTarget` will always be null here?

```suggestion
				var anchor = Popup;
```

Reply:
Now the logic is changed a bit to work around `DatePickerFlyout`, so had to keep this in the end.